crates/rover-client/src/operations/subgraph/check_workflow/runner.rs

```diff
@@ -305,6 +324,51 @@ fn get_lint_response_from_result(
     }
 }
 
+fn get_proposals_response_from_result(
+    target_url: Option<String>,
+    task_status: Option<CheckWorkflowTaskStatus>,
+    task: Option<SubgraphCheckWorkflowQueryGraphCheckWorkflowTasksOnProposalsCheckTask>,
+) -> Option<ProposalsCheckResponse> {
+    match task {
+        Some(result) => {
+            let mut related_proposals = Vec::with_capacity(result.related_proposal_results.len());
+            for proposal in result.related_proposal_results {
+                let status = match proposal.status_at_check {
+                    subgraph_check_workflow_query::ProposalStatus::APPROVED => "APPROVED",
+                    subgraph_check_workflow_query::ProposalStatus::CLOSED => "CLOSED",
+                    subgraph_check_workflow_query::ProposalStatus::DRAFT => "DRAFT",
+                    subgraph_check_workflow_query::ProposalStatus::IMPLEMENTED => "IMPLEMENTED",
+                    subgraph_check_workflow_query::ProposalStatus::OPEN => "OPEN",
+                    _ => "OTHER",
+                };
+                related_proposals.push(RelatedProposal {
+                    status: status.to_string(),
+                    display_name: proposal.proposal.display_name,
+                })
+            }
```

Comment:
what you have is totally reasonable but a `map` statement is more idiomatic.

additionally you can add a `use subgraph_check_workflow_query::ProposalStatus` import at the top to make this a bit more readable.

```suggestion
            let related_proposals: Vec<RelatedProposal> = result.related_proposal_results.iter().map(|proposal| {
                let status = match proposal.status_at_check {
                    ProposalStatus::APPROVED => "APPROVED",
                    ProposalStatus::CLOSED => "CLOSED",
                    ProposalStatus::DRAFT => "DRAFT",
                    ProposalStatus::IMPLEMENTED => "IMPLEMENTED",
                    ProposalStatus::OPEN => "OPEN",
                    _ => "OTHER",
                };
                RelatedProposal {
                    status: status.to_string(),
                    display_name: proposal.proposal.display_name,
                }
            }).collect();
```

Reply:
Thanks! I have made this update